This was a warning in GCC 13. For GCC 14, it is now an error:

../lisp-debug.c: In function 'debug_show_fpu':
../lisp-debug.c:1184:10: error: assignment to 'struct _libc_xmmreg *' from incompatible pointer type 'struct <anonymous> *' [-Wincompatible-pointer-types]
 1184 |     xmmp = &(xp->uc_mcontext.fpregs->_xmm[0]);
      |          ^

--- a/lisp-kernel/lisp-debug.c
+++ b/lisp-kernel/lisp-debug.c
@@ -1181,7 +1181,7 @@ debug_show_fpu(ExceptionInformation *xp, siginfo_t *info, int arg)
 
 #ifdef LINUX
   if (xp->uc_mcontext.fpregs)
-    xmmp = &(xp->uc_mcontext.fpregs->_xmm[0]);
+    xmmp = (struct _libc_xmmreg *) &(xp->uc_mcontext.fpregs->_xmm[0]);
   else
     /* no fp state, apparently */
     return debug_continue;
